index.js

```diff
@@ -45,51 +47,184 @@ const player = {
   ],
   playlists: [
     { id: 1, name: 'Metal', songs: [1, 7, 4] },
-    { id: 5, name: 'Israeli', songs: [4, 5] },
+    { id: 5, name: 'Israeli', songs: [4 , 5] },
   ],
   playSong(song) {
-    console.log(/* your code here */)
+    console.log(`Playing ${song.title} from ${song.album} by ${song.artist} | ${durationFormat(song.duration)}.`)
   },
 }
 
+function parseDuration(duration){
+  const durationArr = duration.split(":");
+  const minDuration = Number(durationArr[0])*60;
+  const secDuration = Number(durationArr[1]);
+  return minDuration + secDuration;
+}
+
+ function throwIfNoPlaylist(id){  
+   if ( !player.playlists.some( (playlist) => playlist.id === id) ) throw "non-existent ID";
+ }
+ function throwIfNoSong(id){
+  if ( !player.songs.some( (song) => song.id === id) ) throw "non-existent ID";
+ }
+ 
+function durationFormat (secDuration) {
+
+  let seconds = secDuration % 60;
+  const fomatedSec = seconds.toString().length === 1 ? "0" + seconds : seconds;
+
+  let minutes = Math.floor(secDuration / 60);
+  const fomatedMin = minutes.toString().length === 1 ? "0" + minutes : minutes;
+
+  return (fomatedMin + ":" + fomatedSec);
+}
+
 function playSong(id) {
-  // your code here
+  for (i = 0; i < player.songs.length; i++) {
+    const song = player.songs[i];
+    if (song.id === id){ 
+      player.playSong(song);
+      return
+    }
+  }
+  throw "non-existent ID"
 }
 
+
 function removeSong(id) {
-  // your code here
+  throwIfNoSong(id)
+  player.songs = player.songs.filter( (song) => song.id !== id);
+  player.playlists = player.playlists.map((playlist) => {
+    return {
+      ...playlist,
+      songs: playlist.songs.filter((song) => song !== id)
+    };
+
+  });
 }
 
+
 function addSong(title, album, artist, duration, id) {
-  // your code here
+  if ( player.songs.some( (song) => song.id === id) ) throw "ID already taken"; 
+  let maxId = 0;
+   player.songs.forEach((song) => {
+    maxId = Math.max(song.id,maxId);     
+   })
+
+  const newId = id ?? maxId + 1;
+  player.songs = [
+    ...player.songs,
+    {
+      id: newId,
+      title : title,
+      album : album,
+      artist : artist,
+      duration : parseDuration(duration) ,
+    }
```

Comment:
you can do this like that:
{
      id: newId,
      title ,
      album ,
      artist ,
      duration : parseDuration(duration) ,
    }